chat/models.py

```diff
@@ -0,0 +1,27 @@
+from django.db import models
+from django.utils import timezone
+
+
+# class ChatManager(models.Manager):
+#     def get_group_chat(self, group_id):
+#         return self.filter(group_id=group_id).order_by('date_time')
+#
+#     def get_student_course_chat(self, student_course_id):
+#         return self.filter(student_course_id=student_course_id).order_by('date_time')
+
+
+class Message(models.Model):
+    message_id = models.BigAutoField(primary_key=True)
+    # sender_id = models.ForeignKey("User", on_delete=models.CASCADE)
+    # student_course_id = models.ForeignKey("Student_Course", on_delete=models.CASCADE)
+    # group_id = models.ForeignKey("Group", on_delete=models.CASCADE)
+    date_time = models.DateTimeField(default=timezone.now,  blank=False)
+    message = models.CharField(max_length=255,  blank=False)
+    # objects = ChatManager()
+
+    # def save(self, *args, **kwargs):
```

Comment:
I've been reading about it and its usually not recommended to do the validation logic inside the save method. 

There is a specific method called `clean` that gets invoked  by Django validation-system and that's where the validation logic is supposed to be. 

however, the method `clean`  is only automatically invoked when the instance is created by `ModelForm` or `ModelSerializer`. meaning that when creating an instance programmatically, it is needed to call `clean` manually (although I believe chat messages will be always created via a form anyway)

I hope I understood this feature correctly